### PR TITLE
removes permeability, rolling it into bio armor

### PR DESCRIPTION
**Pull request by Fikou**
## About The Pull Request

Alternative to #66721
refactors our disease code a tiny bit
removes permeability_coefficient variable from clothing, it decided how much stuff like chems or disease passed through your clothes, while BIO armor only decided how much you could spread diseases yourself, making it pretty much laughable
permeability_coefficient is now fully rolled into bio armor, so your bio protecting stuff will now protect you from other biological hazards like blobs

image
image

videopresentation of it working: https://streamable.com/qhow73

## Why It's Good For The Game

replaces a weird variable with simpler to understand armor, generalizes bio armor as the standard armor against biohazards.

## Changelog

:cl:
balance: some items may protect more or less against diseases now
/:cl:

### CODE REVIEW

**Comment by Colovorat** [comment]
>permeability

i really hate english sometimes

**Comment by Ryll-Ryll** [comment]
I'm sti not sold on the benefit of bio armor

**Comment by tralezab** [comment]
>mini vacationhulen — Today at 6:55 PM
@Ryll-Ryll i'm gonna merge the permeability coeff pr, i agree that bio armor is bad but there is no world where permeability coeff should exist as an overcomplicated mechanic that is at the least worse than bio armor

File: code/datums/components/infective.dm [hunk 1]
```diff
@@ -97,18 +97,18 @@
 /datum/component/infective/proc/try_infect_equipped(datum/source, mob/living/L, slot)
 	SIGNAL_HANDLER
 
-	var/old_permeability
+	var/old_bio_armor
 	if(isitem(parent))
-		//if you are putting an infective item on, it obviously will not protect you, so set its permeability high enough that it will never block ContactContractDisease()
+		//if you are putting an infective item on, it obviously will not protect you, so set its bio armor low enough that it will never block ContactContractDisease()
 		var/obj/item/I = parent
```
Comment (tralezab) [inline code/datums/components/infective.dm:103]: if you use the var you gotta give it a proper name pardner

File: code/datums/components/infective.dm [hunk 2]
```diff
@@ -97,18 +97,18 @@
...
-		old_permeability = I.permeability_coefficient
-		I.permeability_coefficient = 1.01
+		old_bio_armor = I.armor.getRating(BIO)
+		I.armor.setRating(bio = -1)
 
 	try_infect(L, slot2body_zone(slot))
 
 	if(isitem(parent))
 		var/obj/item/I = parent
```
Comment (tralezab) [inline code/datums/components/infective.dm:110]: same as above, use the var = give it a good name

File: code/datums/diseases/_MobProcs.dm [hunk 3]
```diff
@@ -32,83 +32,67 @@
...
-		var/mob/living/carbon/human/M = src
-		switch(target_zone)
-			if(BODY_ZONE_HEAD)
-				if(M.wear_mask && isobj(M.wear_mask))
-					Cl = M.wear_mask
-					passed = prob((Cl.permeability_coefficient*100) - 1)
-
-	else if(ishuman(src))
+	if(ishuman(src))
 		var/mob/living/carbon/human/H = src
```
Comment (tralezab) [inline code/datums/diseases/_MobProcs.dm:65]: use the var = give it a good name

File: code/datums/diseases/_MobProcs.dm [hunk 4]
```diff
@@ -32,83 +32,67 @@
...
-					passed = prob((Cl.permeability_coefficient*100) - 1)
+					passed = prob(100-H.shoes.armor.getRating(BIO))
 
 	if(passed)
-		D.try_infect(src)
+		disease.try_infect(src)
 
 /mob/living/proc/AirborneContractDisease(datum/disease/D, force_spread)
-	if( ((D.spread_flags & DISEASE_SPREAD_AIRBORNE) || force_spread) && prob((50*D.permeability_mod) - 1))
+	if( ((D.spread_flags & DISEASE_SPREAD_AIRBORNE) || force_spread) && prob((50*D.spreading_modifier) - 1))
```
Comment (tralezab) [inline code/datums/diseases/_MobProcs.dm:95]: D > disease like you did above

**Review by tralezab — CHANGES_REQUESTED** [review]
minor things which is convenient though because it gives ryll and i time to BROOD over the pr

**Pull request merged**